### PR TITLE
enable NDP aliveness thread and add recovery for NDP

### PR DESCRIPTION
**Pull request by abhishek-samsung**

### CODE REVIEW

File: os/drivers/audio/ndp120_voice.c [hunk 1]
```diff
@@ -505,8 +509,20 @@ static int ndp120_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct ap
 			break;
 		}
 	}
+
+	if (ret != OK) {
```
Comment (Taejun-Kwon) [inline os/drivers/audio/ndp120_voice.c:513]: hmm I think below logic is required.
```
static void *ndp120_app_device_health_check(void)
{
	if (ndp_120_died) {
		dev->alive = false;
		reset_reload;
	}
	...
}

void reset_reload()
{
	do_reset_reload;
	dev->alive = true;
}

int ndp120_extract_audio(struct ndp120_dev_s *dev, struct ap_buffer_s *apb)
{
	if (!dev->alive) {
		return SYNTIANT_NDP_ERROR_UNINIT;
	}
	...
}

static int ndp120_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct ap_buffer_s *apb)
{
	...
	int ret = ndp120_extract_audio(priv, apb);
	/* Regardless result, dequee from pendq */
		for (tmp = (sq_entry_t *)sq_peek(&priv->pendq); tmp; tmp = sq_next(tmp)) {
		if (tmp == (sq_entry_t *)apb) {
			sq_rem(tmp, &priv->pendq);
			audvdbg("found the apb to remove 0x%x\n", tmp);
			break;
		}
	}
	
	if (ret == SYNTIANT_NDP_ERROR_UNINIT) {
		priv->dev.upper(priv->dev.priv, AUDIO_CALLBACK_UNREACHABLE, NULL, OK);
	}
	...
}
```

Comment (abhishek-samsung) [reply os/drivers/audio/ndp120_voice.c]: Yes, thank you. This makes more sense for our case. I have updated it.

**Review by Taejun-Kwon — DISMISSED** [review]

File: os/drivers/ai-soc/ndp120/src/ndp120_api.c [hunk 2]
```diff
@@ -984,7 +1009,7 @@ ndp120_app_device_health_check(void *d)
 			printf("Error: %d in check_firmware_aliveness\n", s);
 			goto out;
 		}
-		pm_sleep(5000);
+		pm_sleep(3000);
```
Comment (r-prabu) [inline os/drivers/ai-soc/ndp120/src/ndp120_api.c:1012]: What is this magic number about? Can we have config for this?

File: os/drivers/ai-soc/ndp120/src/ndp120_api.c [hunk 3]
```diff
@@ -937,6 +937,8 @@ void ndp120_show_debug(int include_spi)
 	}
 }
 
+int ndp120_init(struct ndp120_dev_s *dev, bool reinit);
```
Comment (r-prabu) [inline os/drivers/ai-soc/ndp120/src/ndp120_api.c:940]: shouldn't this be at declaration section at the top?

File: os/drivers/ai-soc/ndp120/src/ndp120_api.c [hunk 4]
```diff
@@ -1012,29 +1037,33 @@ int ndp120_init(struct ndp120_dev_s *dev)
...
-		auddbg("failed to initialize mb sync mutex variable\n");
-	}
+	if (!reinit) {
 
-	s = pthread_mutex_init(&dev->ndp_mutex_mcu_mb_in, NULL);
-	if (s) {
-		auddbg("failed to initialize mb in mutex variable\n");
-	}
+		s = pthread_mutex_init(&dev->ndp_mutex_mbsync, NULL);
+		if (s) {
```
Comment (r-prabu) [inline os/drivers/ai-soc/ndp120/src/ndp120_api.c:1043]: Is it fine to proceed with the initialization if there is failure in the middle?

File: os/drivers/ai-soc/ndp120/src/ndp120_api.c (hunk 3 above)
Comment (abhishek-samsung) [reply os/drivers/ai-soc/ndp120/src/ndp120_api.c]: Done

File: os/drivers/ai-soc/ndp120/src/ndp120_api.c (hunk 2 above)
Comment (abhishek-samsung) [reply os/drivers/ai-soc/ndp120/src/ndp120_api.c]: Done

File: os/drivers/ai-soc/ndp120/src/ndp120_api.c (hunk 4 above)
Comment (abhishek-samsung) [reply os/drivers/ai-soc/ndp120/src/ndp120_api.c:1049]: This is not related to this PR. You are right, initialization should fail incase mutex_init fails. Let me add it in a new PR along with the comments from #6488 .

**Pull request merged**